### PR TITLE
compiler tab style fixes. language selector ui behaviour 

### PR DESCRIPTION
**Pull request by LianaHus**
fixes https://github.com/ethereum/remix-ide/issues/2100

### CODE REVIEW

**Comment by LianaHus** [comment]
https://github.com/ethereum/remix-ide/pull/2087

File: src/app/tabs/compileTab/compilerContainer.js [hunk 1]
```diff
@@ -294,6 +305,16 @@ class CompilerContainer {
     }
   }
 
+  _updateLanguageSelector () {
+    if (this._retriveVersion() < '0.5.7') {
```
Comment (yann300) [inline src/app/tabs/compileTab/compilerContainer.js]: how this can work? you should probably use semver here.

**Comment by yann300** [comment]
some infos about semver https://www.npmjs.com/package/semver

File: src/app/tabs/compileTab/compilerContainer.js (hunk 1 above)
Comment (LianaHus) [reply src/app/tabs/compileTab/compilerContainer.js]: :) it works. ok I'll look at it

File: src/app/tabs/compileTab/compilerContainer.js [hunk 2]
```diff
@@ -240,7 +248,9 @@ class CompilerContainer {
 
   onchangeLanguage (event) {
     this.compileTabLogic.setLanguage(event.target.value)
-    this.compile()
```
Comment (yann300) [inline src/app/tabs/compileTab/compilerContainer.js]: all the other `on*` method, just trigger a new compilation. Might be more consistent to to keep the same behavior here

File: src/app/tabs/compileTab/compilerContainer.js [hunk 3]
```diff
@@ -110,6 +110,11 @@ class CompilerContainer {
     return el
   }
 
+  _retriveVersion () {
```
Comment (yann300) [inline src/app/tabs/compileTab/compilerContainer.js]: typo

File: src/app/tabs/compileTab/compilerContainer.js (hunk 2 above)
Comment (LianaHus) [reply src/app/tabs/compileTab/compilerContainer.js]: but there was a issue here. build shouldn't be triggered if there is no autocompile an selected file

Comment (LianaHus) [reply src/app/tabs/compileTab/compilerContainer.js]: or I can put the check in the compile() function

Comment (yann300) [reply src/app/tabs/compileTab/compilerContainer.js]: > but there was a issue here. build shouldn't be triggered if there is no autocompile an selected file

This should done in another PR and all the `on*` function should be updated.

Comment (LianaHus) [reply src/app/tabs/compileTab/compilerContainer.js]: ok will create a new PR

**Comment by yann300** [comment]
Click on `Solidity` env (from the front page), the selected compiler is `0.5.1`, the dropdown list is still enabled. I guess it should not be

**Pull request merged**